### PR TITLE
refactored the code for multi level options

### PR DESCRIPTION
**Pull request by sanjaymusale**

### CODE REVIEW

File: src/helper.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+
+const findParent = (item, selectedOption, category, tree, path, callback) => {
+  for (let i = tree.length - 1; i >= 0; i--) {
+    (function () {
+      var currentPath = path.slice();
```
Comment (Kameshwaran) [inline src/helper.js]: What is the use of this `slice`

File: src/helper.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          })]
+          recur(optionsData[i].options, data)
+        }
+      }
+    }
+  }
+  return data;
+}
+
+export default findParent;
```
Comment (Kameshwaran) [inline src/helper.js]: It is quite difficult to understand without proper nomenclature.

File: src/index.js [hunk 3]
```diff
@@ -128,58 +157,94 @@ class MultiLevelSelect extends React.Component {
...
+  }
+
+  optionChecked = (values, optionValue, parent) => {
+    return values.some(e => {
+      if (e.value === parent) {
+        return e.options.some(item => {
+          if (item.value === optionValue) {
+            return true
+          }
+        })
```
Comment (Kameshwaran) [inline src/index.js]: this condition can be refactored like,

```
return e.options.some(item => item.value === optionValue)
```

**Comment by Kameshwaran** [comment]
@sanjaymusale Can you refactor the code? and Apply proper nomenclature?

**Comment by sanjaymusale** [comment]
> nomenclature

okay

File: src/helper.js (hunk 1 above)
Comment (sanjaymusale) [reply src/helper.js]: it creates a copy of path in each loop, if slice is not given it overrides with all the previous values along with new values.

**Pull request merged**